src/rgw/rgw_rest_pubsub.cc

```diff
@@ -289,6 +289,12 @@ class RGWPSCreateTopicOp : public RGWOp {
     ret = ps.get_topic(this, topic_name, result, y, nullptr);
     if (ret == -ENOENT) {
       // topic not present
+
+      // initialize the persistent queue's location. this cannot change for
+      // existing topics. use ':' as the namespace delimiter because its
+      // inclusion in a TopicName would break ARNs
+      dest.persistent_queue = string_cat_reserve(
```

Comment:
just a remonder to add a tracker for topic name validation

Reply:
https://tracker.ceph.com/issues/65212